source/Octopus.Client/OctopusAsyncClient.cs

```diff
@@ -598,16 +597,16 @@ private async Task<T> ReadResponse<T>(HttpResponseMessage response)
             }
         }
 
-        private void ValidateSpaceId(string spaceId)
+        private void ValidateSpaceId(SpaceResource space)
         {
-            if (string.IsNullOrEmpty(spaceId))
+            if (space == null)
             {
-                throw new ArgumentException("spaceId cannot be null");
+                throw new ArgumentNullException(nameof(space));
             }
 
-            if (spaceId == MixedScopeConstants.AllSpacesQueryStringParameterValue)
+            if (space.Id == MixedScopeConstants.AllSpacesQueryStringParameterValue)
```

Comment:
I don't think this `All` parameter check is required anymore since we are passing in a proper `SpaceResource`.

Reply:
fixed